patches/system-appearance-change.patch

```diff
@@ -0,0 +1,219 @@
+ src/nsfns.m  | 11 +++++++-
+ src/nsterm.m | 77 ++++++++++++++++++++++++++++++++++++++++++++++++----
+ 3 files changed, 85 insertions(+), 10 deletions(-)
+
+diff --git a/src/frame.h b/src/frame.h
+index 476bac67fa..82f2654347 100644
+--- a/src/frame.h
++++ b/src/frame.h
+@@ -69,9 +69,10 @@ #define EMACS_FRAME_H
+ #ifdef NS_IMPL_COCOA
+ enum ns_appearance_type
+   {
+-    ns_appearance_system_default,
+-    ns_appearance_aqua,
+-    ns_appearance_vibrant_dark
++   ns_appearance_system_default,
++   ns_appearance_aqua,
++   ns_appearance_vibrant_dark,
++   ns_appearance_dark_aqua
+   };
+ #endif
+ #endif /* HAVE_WINDOW_SYSTEM */
+diff --git a/src/nsfns.m b/src/nsfns.m
+index 273fb5f759..74dbf63616 100644
+--- a/src/nsfns.m
++++ b/src/nsfns.m
+@@ -1269,14 +1269,23 @@ Turn the input menu (an NSMenu) into a lisp list for tracking on lisp side.
+   store_frame_param (f, Qundecorated, FRAME_UNDECORATED (f) ? Qt : Qnil);
+ 
+ #ifdef NS_IMPL_COCOA
++#ifndef NSAppKitVersionNumber10_14
++#define NSAppKitVersionNumber10_14 1671
++#endif
+   tem = gui_display_get_arg (dpyinfo, parms, Qns_appearance, NULL, NULL,
+                              RES_TYPE_SYMBOL);
++
+   if (EQ (tem, Qdark))
+-    FRAME_NS_APPEARANCE (f) = ns_appearance_vibrant_dark;
++    if (NSAppKitVersionNumber >= NSAppKitVersionNumber10_14) {
++       FRAME_NS_APPEARANCE (f) = ns_appearance_dark_aqua;
++    } else {
++      FRAME_NS_APPEARANCE (f) = ns_appearance_vibrant_dark;
++    }
+   else if (EQ (tem, Qlight))
+     FRAME_NS_APPEARANCE (f) = ns_appearance_aqua;
+   else
+     FRAME_NS_APPEARANCE (f) = ns_appearance_system_default;
++
+   store_frame_param (f, Qns_appearance,
+                      (!NILP (tem) && !EQ (tem, Qunbound)) ? tem : Qnil);
+ 
+diff --git a/src/nsterm.m b/src/nsterm.m
+index a8f7540ea2..d9c6808793 100644
+--- a/src/nsterm.m
++++ b/src/nsterm.m
+@@ -2215,12 +2215,21 @@ so some key presses (TAB) are swallowed by the system.  */
+   if (NSAppKitVersionNumber < NSAppKitVersionNumber10_10)
+     return;
+ 
+-  if (EQ (new_value, Qdark))
+-    FRAME_NS_APPEARANCE (f) = ns_appearance_vibrant_dark;
+-  else if (EQ (new_value, Qlight))
++  if (EQ (new_value, Qdark)) {
++#if MAC_OS_X_VERSION_MAX_ALLOWED >= 101400
++#ifndef NSAppKitVersionNumber10_14
++#define NSAppKitVersionNumber10_14 1671
++#endif
++    if (NSAppKitVersionNumber >= NSAppKitVersionNumber10_14)
++      FRAME_NS_APPEARANCE(f) = ns_appearance_dark_aqua;
++    else
++#endif /* MAC_OS_X_VERSION_MAX_ALLOWED >= 101400 */
++      FRAME_NS_APPEARANCE(f) = ns_appearance_vibrant_dark;
++  } else if (EQ(new_value, Qlight)) {
+     FRAME_NS_APPEARANCE (f) = ns_appearance_aqua;
+-  else
++  } else {
+     FRAME_NS_APPEARANCE (f) = ns_appearance_system_default;
++  }
+ 
+   [window setAppearance];
+ #endif /* MAC_OS_X_VERSION_MAX_ALLOWED >= 101000 */
+@@ -7595,7 +7604,6 @@ - (instancetype) initFrameFromEmacs: (struct frame *)f
+   if (! FRAME_UNDECORATED (f))
+     [self createToolbar: f];
+ 
+-
```

Comment:
Ok, it seems that the conflict is trivial and all you need to do is to remove this line from patch. I will fix that by myself. Don't worry :)